### DISCUSSION TITLE
Is there a auth flow, which allows a client to login without entering the master password?

### DISCUSSION THREAD

**Discussion by dakky** [dani-garcia/vaultwarden#4004] [General]
I know there a auth mechanisms, which allow a device to log in without entering the password of the account itself. Nextcloud i.e. uses this. 

Basically this works this way:
1. app/client wants access to account
2. you need to login in the webui of the service
3. you get a promt in the webui if this access of the client should be validated and allowed
4. if yes, a internal device password/keypair is generated and used ONLY for this device

why do i ask?
* security concerns: 
  I do not trust my mobile enough, to enter the plain text master password of my vault. It might be stolen => need to change master password on all used locations instead of revoking the single device password
* usability: 
   my master password is about 35 bytes containing the full bandwith of valid characters. Typing this on a keyboard works fine thanks to my muscle memory :smile:  typing this on a screen is nearly impossible. And I refuse to lower the security of my vault by going to a simpler password

**Comment by BlackDex** [comment]
So, you want to login without a master password to make it less secure?
That would be bad in my opinion.

But basically, no. The master-password is needed to decrypt the keys/vault.
Via SSO there are some options, but that needs an organization and a key connector service. That way login via SSO is possible, though, i do not know if that can be done via mobile devices in a correct way.
But besides that, it is not (yet) supported by Vaultwarden.

**Reply by BlackDex** [reply]
That said, if you can use Biometrics or PIN to just login in once, and use Biometrics or PIN to unlock the vault, but not login.

**Reply by dakky** [reply]
Hello @BlackDex Thanks for the fast response.

> So, you want to login without a master password to make it less secure?

THAT is diskussable :laughing:  basically the device password might be as complex/hard as possible.  Imagine something like a 4096bit key ... waaaay more secure than my master password.

But yes of course: I get your point: the key to decrypt the vault must come to the device. the one way or another.

> That said, if you can use Biometrics or PIN

Hmm Biometrics might be a possible solution. Didn't know this is possible. So basically I have to go ONE time through the process of entering my master key and this one will be locked on the device with my fingerprint?
